element-connector/src/main/java/org/eclipse/californium/elements/RelaxedCorrelationContextMatcher.java

```diff
@@ -36,29 +36,19 @@ public boolean isResponseRelatedToRequest(CorrelationContext requestContext, Cor
 		return internalMatch(requestContext, responseContext);
 	}
 
+	@Override
+	public boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext) {
+		return internalMatch(messageContext, connectorContext);
+	}
+
 	private final boolean internalMatch(CorrelationContext requestedContext, CorrelationContext availableContext) {
 		if (null == requestedContext) {
 			return true;
 		} else if (null == availableContext) {
 			return false;
 		}
 		if (requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
-			boolean match = requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID).equals(
-					availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID))
-					&& requestedContext.get(DtlsCorrelationContext.KEY_CIPHER).equals(
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER));
-
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS session {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID),
-							availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID) });
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS cipher {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_CIPHER),
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER) });
-			return match;
+			return CorrelationContextUtil.match(getName(), KEYS, requestedContext, availableContext);
```

Comment:
I tried to sum-up the information about the correlation context and it's usage:

Information sum-up:

#142 
asymetric processing of requests and responses correlation context, potential extension of correlation context with information not intended to be used for "matching"

#145
UserContext and CorrelationContext, CorrelationContext is californium internal

#156 
Correlation Context is defined by keys (and their associated value) and a "secure" message uses the key "DtlsCorrelationContext.KEY_SESSION_ID".

Current usage:
Only DtlsCorrelationContext (and TcpCorrelationContext) is used. Therefore either no correlation context is provided, or a DtlsCorrelationContext. Therefore the "else ... equals" was not used.

So my conclusion:

Generally the first part of the matching:

```
        if (exchange.getCorrelationContext() == null) {
		// no correlation information available for request, thus any
		// additional correlation information available in the response is ignored
		return true;
	}
```
applies to "all" correlation contexts.

The second test, is currently base on DtlsCorrelationContext.KEY_SESSION_ID. I think, that matching should be defined by matching equal values for a set of keys. If null for both values of a key are also considered to be matching, the keys set may just be the super set over different correlation context types (e.g. TcpCorrelationContext.KEY_CONNECTION_ID could also be added. The result would be, that either the DTLS information is matched (provided by DtlsCorrelationContext) or the tcp information (provided by TcpCorrelationContext)). 

The third test 

```
        } else {
		// compare message context used for sending original request to context
		// the response has been received in
		return exchange.getCorrelationContext().equals(responseContext);
	}

```
should be removed

But I would call the "matcher" for all connectors, not only the "secure" ones.